### PR TITLE
Feature/hp 233 add infrastructure

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

**Review by vasiariabov — APPROVED** [review]
now do I need to move sentry to consul for all environments?

**Comment by hasmikharutyunyan** [comment]
> now do I need to move sentry to consul for all environments?

Yes, after merge. I think  we can wait approve of @DFSko

**Review by oleg-n-k — APPROVED** [review]

File: HappyTravel.Nakijin.Api/Infrastructure/ServicesConfigurationExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+        });
+
+        builder.Services.AddSwagger();
+
+        builder.Services.AddMvcCore()
+            .AddFormatterMappings()
+            .AddNewtonsoftJson()
+            .AddApiExplorer()
+            .AddCacheTagHelper()
+            .AddControllersAsServices()
```
Comment (DFSko) [inline HappyTravel.Nakijin.Api/Infrastructure/ServicesConfigurationExtensions.cs:92]: do we need it? do we really inject controllers as dependency somewhere?

File: HappyTravel.Nakijin.Api/Infrastructure/AppConfigurationExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace HappyTravel.Nakijin.Api.Infrastructure;
+
+public static class AppConfigurationExtensions
+{
+    public static void Configure(this WebApplication app)
+    {
+             app.UseProblemDetailsErrorHandler(app.Environment, app.Logger);
+             app.UseHttpContextLogging(
+                 options => options.IgnoredPaths = new HashSet<string> { "/health" }
```
Comment (DFSko) [inline HappyTravel.Nakijin.Api/Infrastructure/AppConfigurationExtensions.cs:18]: maybe add "/metrics"? prometheus fetch metrics every 5 seconds

File: HappyTravel.Nakijin.Updater/Infrastructure/ServicesConfigurationExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+        services.AddTransient<LocationsCollector>();
+        */
+
+        builder.Services.AddStaticDataPublicationServices(vaultClient, builder.Configuration, builder.Environment);
+        builder.Services.AddHostedService<DataUpdateHostedService>();
+    }
+
+
+    private static IServiceCollection AddServices(this IServiceCollection services)
```
Comment (DFSko) [inline HappyTravel.Nakijin.Updater/Infrastructure/ServicesConfigurationExtensions.cs:84]: Why not use WebApplicationBuilder and builder.Services and builder.Configuration?

**Review by DFSko — APPROVED** [review]

**Pull request merged**